Project proposal for dark mode project (#3913)
WordPress/openverse: documentation/projects/proposals/dark_mode/20240313-project_proposal_dark_mode.md

# 2024-03-13 Project Proposal

**Author**: @fcoveram

## Reviewers

<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->

> - [ ] @obulat

Reviewer: ```suggestion
- [x] @obulat
```

> - [ ] @zackkrida

Reviewer 2: ```suggestion
- [x] @zackkrida
```

# 2024-03-13 Project Proposal

**Author**: @fcoveram

## Reviewers

<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->

- [ ] @obulat
- [ ] @zackkrida

## Project summary

<!-- A brief one or two sentence summary of the project's features -->

Include a feature that changes the UI from light to a dark color scheme.

### Motivation

Dark mode is a customization level that is in the spirit of adapting the UI to

> people’s contexts. The benefits span a11y, personal preferences set on the

Reviewer 2: ```suggestion
user's contexts. The benefits span a11y, personal preferences set on the
```

Small nit-pick, "user" might be clearer.

devices, energy efficiency in devices, and an alternative browsing experience of
visual content due to the background contrast.

## Goals

<!-- Which yearly goal does this project advance? -->

Refine Search Experience

> ## Requirements

Reviewer 2: In addition to the design documentation requirements you listed, this section should include the requirements of the dark mode functionality itself. Some possible examples:

- The user can toggle dark,light, or system mode through an easy-to-find user interface control
- The location of that UI control (header? footer?)
- What the default value should be (light or system?)
- Goals for the dark mode itself:
  - Preserve the visual identity of Openverse, sufficient contrast, etc. 

The goal of this section of the plan is that if, hypothetically, another designer had to take on this work, they would have a good understanding of your intentions.

Author: I am not sure about the second point as it specifies an UI decision that can be discussed on the design ticket once is created. The first point covers the design approach needed more broadly and gives room to design iteration.

For the remaining points:
* The default setting is something I missed, thanks for surfacing it.
* I will add the color contrast as an a11y requirement.
* Preserving the brand identity is a great idea, adding it.

Thanks for the suggestions.

<!-- Detailed descriptions of the features required for the project. Include user stories if you feel they'd be helpful, but focus on describing a specification for how the feature would work with an eye towards edge cases. -->

1. Design documentation of pages and UI components in the Design Library and any
   other additional Figma files.
2. Design documentation of transition from current to new frontend
   implementation.

## Success

<!-- How do we measure the success of the project? How do we know our ideas worked? -->

> 
> - Add the dark theme to the site by allowing visitors to change the UI through
>   three options: Light theme, Dark theme, or follow device setting.
> - Document the dark variant of every UI element in both the Design Library and
>   Storybook.

Reviewer 2: ```suggestion

- Add the dark theme to the site by allowing visitors to change the UI through
  three options: Light theme, Dark theme, or follow device setting.
- Document the dark variant of every UI element in both the Design Library and
  Storybook.
```

I think these are more "requirements" of the project, rather than ways of measuring success. How would we confirm that adding this functionality was beneficial? Perhaps an analytics event on the theme switcher, user feedback, etc?

Author: You are right. Those are requirements.

Regarding success, I don't think that user feedback is necessary for this project. Switching the UI's theme to fit users' preference from the device or site setting feels similar to building a responsive site or designing with a11y in mind. It adapts content to user's context.

It doesn't add, modify, or remove features or content, but it expands the way of how people use Openverse.

The above doesn't imply leaving the section empty, but after stating the above, I would approach the success from a technical perspective. What do you think?

Author: I'm not against of adding it. I will include the action in the section.

## Participants and stakeholders

<!-- Who is working on the project and who are the external stakeholders, if any? Consider the lead, implementers, designers, and other stakeholders who have a say in how the project goes. -->

- Lead: @fcoveram
- Design: @fcoveram
- Implementation: TBD

## Infrastructure

<!-- What infrastructural considerations need to be made for this project? If there are none, say so explicitly rather than deleting the section. -->

> TBD

Reviewer 2: ```suggestion
This project only concerns frontend user interface code, and swapping colors. It should not require any changes to our infrastructure.
```

## Accessibility

<!-- Are there specific accessibility concerns relevant to this project? Do you expect new UI elements that would need particular care to ensure they're implemented in an accessible way? Consider also low-spec device and slow internet accessibility, if relevant. -->

- Follow the advisor of WCAG 2.2 to meet AA. Put special attention to the

>   [Perceivable section](https://www.w3.org/WAI/WCAG22/quickref/?currentsidebar=%23col_overview&levels=aaa&showtechniques=321#principle1).

Reviewer 2: From the docs in the template:

> Are there specific accessibility concerns relevant to this project? **Do you expect new UI elements that would need particular care to ensure they're implemented in an accessible way?** Consider also low-spec device and slow internet accessibility, if relevant.

Is the component you envision for switching color themes one we already have in our component library, or a new control? Could it have any accessibility issues?

Author: > Is the component you envision for switching color themes one we already have in our component library, or a new control?

It already exist in the library.

> Could it have any accessibility issues?

If current checkbox, radio, and toggle components don't have accessibility issues, then everything is fine.

Reviewer 2: Excellent. Can you mention that in the document?

## Marketing

<!-- Are there potential marketing opportunities that we'd need to coordinate with the community to accomplish? If there are none, say so explicitly rather than deleting the section. -->

Since the project involves visual changes across the site, there is a marketing

> opportunity to introduce the benefits of the UI’s dark theme.

Reviewer 2: It would be good to think about this as much as possible now, rather than later. How much would we want to market this? The WordPress account would share socials for us, I am sure. Would we just want to create a screenshot of dark mode? Or a post on the make site showing a more detailed breakdown? Or a little UI tour video? What do you think? As the project lead, you're able to make these choices as to what you think is the best balance of effort vs. outcome.

Author: I like the idea of a short video showing the feature and quick glances of the three search results and the media details page, but sharing this point with [marketing team](https://wordpress.slack.com/archives/C0GKJ7TFA) for more ideas and tap into existing approaches I might not be aware of.

# 2024-03-13 Project Proposal

**Author**: @fcoveram

## Reviewers

<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->

- [ ] @obulat
- [ ] @zackkrida

## Project summary

<!-- A brief one or two sentence summary of the project's features -->

Include a feature that changes the UI from light to a dark color scheme.

### Motivation

Dark mode is a customization level that is in the spirit of adapting the UI to
user's contexts. The benefits span a11y, personal preferences set on the
devices, energy efficiency in devices, and an alternative browsing experience of
visual content due to the background contrast.

## Goals

<!-- Which yearly goal does this project advance? -->

Refine Search Experience

## Requirements

<!-- Detailed descriptions of the features required for the project. Include user stories if you feel they'd be helpful, but focus on describing a specification for how the feature would work with an eye towards edge cases. -->

> 1. Add the dark theme to the site and set the Light theme as default.

Author: Because the current theme is light and the unexpected change is not a good way of introducing the feature. With that in mind, we should include an onboarding process.

2. Allow visitors to change the UI in an reachable manner from the whole site
   through three options: Light theme, Dark theme, or follow device setting.
3. Design a dark theme preserving the brand identity.
4. Design a dark theme that meet the color contrast requirements pointed out in
   the accessibility section below.
5. Document the design of pages and UI components in the Design Library and any
   other additional Figma files.
6. Document the dark variant of every UI element on Storybook.
7. Document the transition from current to new frontend implementation.

## Success

<!-- How do we measure the success of the project? How do we know our ideas worked? -->

> 

Author: That's a good idea. It would still need a human revisit, as some contrast requirements change depending on the element's context.
